String.cxx

```diff
@@ -0,0 +1,127 @@
+    if (s.m_u.size() > m_u.size())
+        return false;
+    return (m_u.compare(m_u.length() - s.m_u.length(), s.m_u.length(), s.m_u) == 0);
+}
+
+void String::getBytes(byte buffer[], unsigned length) const noexcept{
+    std::transform(m_u.begin(), m_u.end(), std::back_inserter(buffer), [](unsigned char c) -> byte { return c; });
+}
+
+[[nodiscard]] int String::indexOf(const char* c) const noexcept { return m_u.find(c); }
+
+[[nodiscard]] int String::indexOf(const char* c, unsigned index) const noexcept { return m_u.find(c, index); }
+
+[[nodiscard]] int String::indexOf(const String& str) const noexcept { return m_u.find(str); }
+
+[[nodiscard]] int String::indexOf(const String& str, unsigned index) const noexcept { return m_u.find(str, index); }
+
+void String::remove(unsigned idx) { m_u.erase(index); }
+
+void String::remove(unsigned idx, unsigned count) { m_u.erase(idx, idx + count - 1); }
+
+void String::replace(const String& substring1, const String& substring2) {
+    size_t position = m_u.find(substring1.m_u);
+
+    while (position != std::string::npos) {
+        m_u.replace(position, m_u.size(), substring2.m_u);
+        position = m_u.find(substring1.m_u, position + substring2.m_u.size());
+    }
+}
+
+void String::reserve(unsigned size) { m_u.reserve(size); }
+
+void String::setCharAt(unsigned index, const char* c) { m_u[index] = c; }
+
+[[nodiscard]] String String::substring(unsigned from) const { return m_u.substr(from); }
+
+[[nodiscard]] String String::substring(unsigned from, unsigned to) const { return m_u.substr(from, to - from); }
+
+void String::toCharArray(const char* buffer[], unsigned length) noexcept {
+    if (length <= m_u.length) {
+        std::memcpy(buffer, m_u, length);
+    } else {
+        std::memcpy(buffer, m_u, m_u.length());
+    }
+}
+
+[[nodiscard]] long String::toInt() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stoi(m_u);
+        } 
+        else return 0;
+}
+
+[[nodiscard]] double String::toDouble() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stod(m_u);
+        } 
+        else return 0;
+}
+
+[[nodiscard]] float String::toFloat() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stof(m_u);
+        } 
+        else return 0;
+}
+
+void String::toLowerCase() noexcept{ std::transform(m_u.begin(), m_u.end(), m_u.begin(), std::tolower); }
+
+void String::toUpperCase() noexcept{ std::transform(m_u.begin(), m_u.end(), m_u.begin(), std::toupper); }
+
+void String::trim() { m_u = m_u.erase(std::remove(m_u.begin(), m_u.end(), ' '), m_u.end()); }
+
+[[nodiscard]] bool String::equals(const String& s) const noexcept { return m_u == s.m_u; }
+
+[[nodiscard]] bool String::equalsIgnoreCase(const String& s) const noexcept {
+    String str = m_u;
+    return (std::transform(s.begin(), s.end(), s.begin(), std::tolower) == std::transform(str.begin(), str.end(), str.begin(), std::tolower));
```

Comment:
This won't compile; you probably meant `std::string str = m_u;`, as well as to copy `s.m_u` in another variable